### PR TITLE
Re-focus view button when closing details panel

### LINKED ISSUE #3362: When we close "Details view" section focus moves to top of the page: A11y_.NetAspireDashboard_Resource_Details view_Keyboard
"[Check out Accessibility Insights](https://nam06.safelinks.protection.outlook.com/?url=https://accessibilityinsights.io/&data=02%7c01%7cv-stfe%40microsoft.com%7cb67b2c4b646d4f9561a208d6f4b5c39b%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c636965458850501301&sdata=mxhokIKNMb22llsjXHLgU3XZibj1Qfx37rpY4PU2sfE%3D&reserved=0)! - Identify accessibility bugs before check-in and make bug fixing faster and easier."

**Please do not close this bug. This bug should only be closed by Trusted Tester after verification.**

### Tags:
#A11yTCS; #A11yMAS; #A11ySev2; #Win11; #ChromiumEdge; #FTP; #DesktopWeb; #.NET Core; #SH_.Net Aspire Dashboard_Web_Mar2024; #WCAG2.4.3; #Keyboard;

### Environment Details:
Application Name: .Net Aspire Dashboard
URL: https://localhost:19888/
Chrome Version 122.0.2365.80 (Official build) (64-bit)
Window Version: Win 11
OS Build: 22621.3085

### Repro Steps:
1. Hit the URL: https://localhost:15887 and login with valid credentials.
2. TAB to "Resources" tab item and hit ENTER key.
3. TAB to "View" button present under 'Details' column header in the page and hit ENTER key.
4. "Details container" will open. TAB to close button on the details view section and hit ENTER key.
5. Observe if focus is back on the view button on the table.

### Actual Results:
When we close "Details view" section focus moves to top of the page.

**Similar issue is observed in throughout the application.**

### Expected Results:
Focus should move back to 'View' button after closing "Details View" section.

### User Impact:
Keyboard users will face difficulty here while navigating inside the page.

### Attachment:
[3362_A11y_.NetAspireDashboard_Resource_Details view_Keyboard.webm](https://github.com/dotnet/aspire/assets/90901519/c0c97d2e-efd7-41c0-8550-7abe21b6ff20)

### PR DESCRIPTION
**Pull request by adamint**
Animation

Fixes #3362 - this will have to be changed when we remove the view buttons
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/3368)

### CODE REVIEW

File: src/Aspire.Dashboard/Components/Pages/StructuredLogs.razor [hunk 1]
```diff
@@ -102,7 +102,10 @@
                                 }
                             </TemplateColumn>
                             <TemplateColumn Title="@ControlsStringsLoc[nameof(ControlsStrings.DetailsColumnHeader)]" Class="no-ellipsis">
-                                <FluentButton Appearance="Appearance.Lightweight" OnClick="() => OnShowProperties(context)">@ControlsStringsLoc[nameof(ControlsStrings.ViewAction)]</FluentButton>
+                                @{
+                                    var id = $"details-button-{context.GetHashCode()}";
```
Comment (JamesNK) [inline src/Aspire.Dashboard/Components/Pages/StructuredLogs.razor:106]: GetHashCode isn't unique. Probably the best thing is to add an InternalId property of guid to the type and have it assigned when the data is received.

File: src/Aspire.Dashboard/Components/Pages/Resources.razor.cs [hunk 2]
```diff
@@ -202,21 +206,28 @@ private bool ApplicationErrorCountsChanged(Dictionary<OtlpApplication, int> newA
...
-    private void ClearSelectedResource()
+    private async Task ClearSelectedResourceAsync()
     {
         SelectedResource = null;
+
+        if (_elementIdBeforeDetailsViewOpened is not null)
+        {
+            await JS.InvokeVoidAsync("focusElement", _elementIdBeforeDetailsViewOpened);
+        }
     }
```
Comment (JamesNK) [inline src/Aspire.Dashboard/Components/Pages/Resources.razor.cs:231]: You've changed this so the view button is always focused when the resource is closed, but there are a number of ways to close the resource. For example, if someone changes search parameters on a page then the resource is closed. We don't want to focus the button in that scenario.

What is the exact scenario where the view button should be focused?

Comment (adamint) [reply src/Aspire.Dashboard/Components/Pages/Resources.razor.cs]: You can check the linked bug, but this is in reaction to the user manually closing the details panel.

> You've changed this so the view button is always focused when the resource is closed

Oops, thanks. Fixed

File: src/Aspire.Dashboard/Components/Pages/Resources.razor.cs [hunk 3]
```diff
@@ -202,21 +206,30 @@ private bool ApplicationErrorCountsChanged(Dictionary<OtlpApplication, int> newA
...
+            await ClearSelectedResourceAsync();
         }
         else
         {
             SelectedResource = resource;
         }
     }
 
-    private void ClearSelectedResource()
+    private async Task ClearSelectedResourceAsync(bool causedByUserAction = false)
```
Comment (tlmii) [inline src/Aspire.Dashboard/Components/Pages/Resources.razor.cs:223]: nit: Technically the other ways it gets dismissed are by user actions too (e.g. filtering). Maybe `fromExplicitDismiss` or something is more accurate?

**Review by tlmii — APPROVED** [review]
> this will have to be changed when we remove the view buttons

We should make sure that we note this somewhere in the issues list. This PR addresses the issue we need to address, so approving from that perspective.

**Pull request merged**